v1/src/test/java/com/google/cloud/teleport/templates/BulkCompressorLT.java

```diff
@@ -0,0 +1,118 @@
+import static com.google.cloud.teleport.it.matchers.TemplateAsserts.assertThatPipeline;
+import static com.google.cloud.teleport.it.matchers.TemplateAsserts.assertThatResult;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.storage.Storage;
+import com.google.cloud.teleport.it.DataGenerator;
+import com.google.cloud.teleport.it.TemplateLoadTestBase;
+import com.google.cloud.teleport.it.TestProperties;
+import com.google.cloud.teleport.it.artifacts.ArtifactClient;
+import com.google.cloud.teleport.it.artifacts.GcsArtifactClient;
+import com.google.cloud.teleport.it.common.ResourceManagerUtils;
+import com.google.cloud.teleport.it.launcher.PipelineLauncher.LaunchConfig;
+import com.google.cloud.teleport.it.launcher.PipelineLauncher.LaunchInfo;
+import com.google.cloud.teleport.it.launcher.PipelineOperator.Result;
+import com.google.cloud.teleport.metadata.TemplateLoadTest;
+import com.google.common.base.MoreObjects;
+import com.google.re2j.Pattern;
+import java.io.IOException;
+import java.text.ParseException;
+import java.time.Duration;
+import java.util.function.Function;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Performance test for {@link BulkCompressor BulkCompressor} template. */
+@TemplateLoadTest(BulkCompressor.class)
+@RunWith(JUnit4.class)
+public class BulkCompressorLT extends TemplateLoadTestBase {
+  private static final String SPEC_PATH =
+      MoreObjects.firstNonNull(
+          TestProperties.specPath(), "gs://dataflow-templates/latest/Bulk_Compress_GCS_Files");
+  private static final String ARTIFACT_BUCKET = TestProperties.artifactBucket();
+  private static final String TEST_ROOT_DIR = BulkCompressorLT.class.getSimpleName().toLowerCase();
+  // 35,000,000 messages of the given schema make up approximately 10GB
+  private static final String NUM_MESSAGES = "35000000";
+  private static ArtifactClient artifactClient;
+
+  @Before
+  public void setup() {
+    // Set up resource managers
+    Storage gcsClient = createStorageClient(CREDENTIALS);
+    artifactClient = GcsArtifactClient.builder(gcsClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+  }
+
+  @After
+  public void teardown() {
+    ResourceManagerUtils.cleanResources(artifactClient);
+  }
+
+  @Test
+  public void testBacklog10gb() throws IOException, ParseException, InterruptedException {
+    testBacklog10gb(Function.identity());
+  }
+
+  public void testBacklog10gb(Function<LaunchConfig.Builder, LaunchConfig.Builder> paramsAdder)
+      throws IOException, ParseException, InterruptedException {
+    DataGenerator dataGenerator =
+        DataGenerator.builderWithSchemaTemplate(testName, "GAME_EVENT")
+            .setQPS("1000000")
+            .setMessagesLimit(NUM_MESSAGES)
+            .setSinkType("GCS")
+            .setOutputDirectory(getTestMethodDirPath() + "/input")
```

Comment:
nit, in ITs we are using `getGcsPath("input")` defined in the base class. can be eventually consolidated (no need to do it now)

Reply:
Sounds good